File: qc-core/src/main/java/com/eager/questioncloud/core/domain/hub/post/implement/PostRemover.java
package com.eager.questioncloud.core.domain.hub.post.implement;

import com.eager.questioncloud.core.domain.hub.post.model.Post;
import com.eager.questioncloud.core.domain.hub.post.repository.PostRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class PostRemover {
    private final PostRepository postRepository;

    public void delete(Long postId, Long userId) {
        Post post = postRepository.findByIdAndWriterId(postId, userId);
        postRepository.delete(post);
    }
}

File: qc-core/qc-post/qc-post-api/src/main/kotlin/com/eager/questioncloud/post/implement/PostRemover.kt
package com.eager.questioncloud.post.implement

import com.eager.questioncloud.post.command.DeletePostCommand
import com.eager.questioncloud.post.repository.PostRepository
import org.springframework.stereotype.Component

@Component
class PostRemover(
    private val postRepository: PostRepository,
) {
    fun delete(command: DeletePostCommand) {
        val post = postRepository.findByIdAndWriterId(command.postId, command.userId)
        postRepository.delete(post)
    }
}